build/config-service.mjs

```diff
@@ -0,0 +1,22 @@
+import path from 'path'
+const projectRoot = process.cwd()
+const webpack = {
+  common: {
+    context: projectRoot,
+    entry: './src/cedict-service/service.js',
+    resolve: {
+      alias: {
+        '@lexisCs': path.join(projectRoot, 'src')
```

Comment:
It uses `@lexisCS` instead of `@` because components import modules not from the transpiled Lexis CS output script but from the Lexis CS code directly. Having it as `@lexisCS` allows us to redefine this alias within components to point to the Lexis CS source folder within node modules. Please see `components#cedict` for an example of that.

This allows us to solve the issue with treeshaking without the use of Rollup (files transpiled by Webpack cannot be treeshaken until version 5 is out, I believe). I think it is a very simple approach that allows to build efficient code without going into too much complexities. What do you think?

Reply:
Let's take `data-models`, for example. When we build it, webpack produces a `dist/alpheios-data-models.js` bundle. There is no import or export statements there, everything is wrapped inside `__webpack_require()` and other webpack specific functions. The generated code looks pretty complicated and I think it is like that.

When we do `import Homonym from 'alpheios-data-models'` in components, for example, in resolving this Webpack goes to `/node_modules/alpheios-data-models/dist/alpheios-data-models.js` (which is a set of webpack specific function wrappers in one bundle), finds the function that returns Homonym, and executes it. Webpack cannot treeshake alpheios-data-models.js as of currently: it will include the whole `alpheios-data-models.js` bundle into the resulting app, even if some functions or classes from it are never used.

When we import individual modules from, let's say, alpheios-data-models directly (as `import Homonym from '@dataModels/homonym.js`) it will import just the Homonym class and nothing more. We'll get a bundle that will take less space, even with the current version of webpack. As a positive side effect, if we're working on components and making changes to data-models that are needed for that, we don't need to build data-models and then components every time; building just components is enough. During the components build process webpack will go directly to the data-models source for `homonym.js` and will pick up the changes.

Reply:
ok, thanks for this explanation. It makes sense to me.